spec/search_history_spec.rb

```diff
@@ -1,15 +1,49 @@
 require 'spec_helper'
 
 RSpec.describe Kitchen::SearchHistory do
+  context 'init' do
+    it 'creates instance variables' do
+      sh = described_class.new('foo', 'bar')
+      expect(sh.upstream).to eq('foo')
+      expect(sh.latest).to eq('bar')
+    end
+  end
+
   context 'empty' do
     it 'converts to an empty array' do
       expect(described_class.empty.to_a).to eq []
     end
   end
 
+  context '#add' do
+    it 'adds nil if given nil value' do
+      expect(described_class.empty.add(nil).to_a).to eq [nil]
+    end
```

Comment:
Nevermind ~Looking at the `[css_or_xpath].join(', ')` part of `#add` I'm confused why I did that.  You might look through the codebase and see if we ever pass in an array as the `css_or_xpath` -- that's the only reason I can think of that we'd need a `join` but at the same time if `css_or_xpath` is an array, then we have nested arrays which is also confusing.~

Reply:
I see that I used `to_a` when testing, but there might be a better way.  Here you could maybe say

```ruby
expect(described_class.empty.add(nil).latest).to be_nil
```